contracts/bridge/contracts/binance/BasicInboundChannel.sol

```diff
@@ -83,18 +85,28 @@ contract BasicInboundChannel {
 
     function processMessages(Message[] memory messages) internal {
         for (uint256 i = 0; i < messages.length; i++) {
+            Message memory message = messages[i];
             // Check message nonce is correct and increment nonce for replay protection
-            require(messages[i].nonce == nonce + 1, "Channel: invalid nonce");
+            require(message.nonce == nonce + 1, "Channel: invalid nonce");
 
             nonce = nonce + 1;
 
-            // Deliver the message to the target
-            (bool success, ) =
-                messages[i].target.call{value: 0, gas: MAX_GAS_PER_MESSAGE}(
-                    messages[i].payload
-                );
-
-            emit MessageDispatched(messages[i].nonce, success);
+            try ICrossChainFilter(message.target).filtered(message.origin) 
```

Comment:
What if some else contract also implement filter() interface by accident.

Reply:
I changed the interface to a less collision-prone name, and the registration pattern requires more complex operations on the part of the app layer.